core/src/main/java/io/varhttp/VarConfigurationContext.java

```diff
@@ -138,10 +138,14 @@ private Object getAndInitializeFilter(Method method, FilterTuple filterTuple) {
 
 	private Object getAndInitializeDefaultFilter(Method method, Class<?> filterClass) {
 		if (Filter.class.isAssignableFrom(filterClass)) {
-			return getObjectFactory().getInstance(filterClass);
+			Filter filter = (Filter) getObjectFactory().getInstance(filterClass);
```

Comment:
What is the difference between filter, varfilter and varfilterexecution

Reply:
VarFilter is a filter that needs to be initialized. Then it either needs:
- to implement Filter interface for the doFilter method
- or use the @FilterMethod annotation that will be picked up by the VarFilterExecution wrapper.

Reply:
maybe at some point we need to unify this because is becoming too confusing. To many ways of using the filters 😬